### PR TITLE
Add support for async stepping

### PR DESCRIPTION
**Pull request by johnmccutchan**
- Require ^0.2.1 of vm_service_libs
- Update isolate's suspended state based on the pause event's atAsyncSuspension field
- Add stepOverAsyncSuspension
- Add autoStepOver
- Wired UI stepOver to use autoStepOver

### CODE REVIEW

**Comment by johnmccutchan** [comment]
This is backwards compatible in that older versions of the VM won't send the "atAsyncSuspension" bit and then the client won't attempt to do a step 'OverAsyncSuspension'

File: pubspec.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 name: atom_dartlang
-version: 0.6.0
+version: 0.6.1
```
Comment (devoncarew) [inline pubspec.yaml]: Can you drop this back down? We have a more complicated dance for rev'ing an atom plugin.

Comment (johnmccutchan) [inline pubspec.yaml]: Done.

File: lib/debug/observatory_debugger.dart [hunk 2]
```diff
@@ -496,7 +505,15 @@ class ObservatoryIsolate extends DebugIsolate {
   stepIn() => service.resume(isolateRef.id, step: StepOption.kInto);
   stepOver() => service.resume(isolateRef.id, step: StepOption.kOver);
   stepOut() => service.resume(isolateRef.id, step: StepOption.kOut);
-
+  stepOverAsyncSuspension() =>
+      service.resume(isolateRef.id, step: StepOption.kOverAsyncSuspension);
```
Comment (devoncarew) [inline lib/debug/observatory_debugger.dart]: Ok, so this will only send `kOverAsyncSuspension` into a newer client, as older clients will cause the `atAsyncSuspension` field to populate w/ false above.

File: lib/debug/observatory_debugger.dart [hunk 3]
```diff
@@ -473,12 +478,16 @@ class ObservatoryIsolate extends DebugIsolate {
       (libraryRef) => new ObservatoryLibrary._(libraryRef)).toList();
   }
 
-  void _suspend(bool value) {
-    if (!value) frames = null;
+  void _suspend(bool paused, bool pausedAtAsyncSuspension) {
```
Comment (devoncarew) [inline lib/debug/observatory_debugger.dart]: Perhaps used a named param for the 2nd param, just to avoid having call sites w/ a list of booleans (`_suspend(false, false)`) - can be hard to parse later.

**Comment by devoncarew** [comment]
lgtm, thanks! One suggestion about using a named param for clarity.

File: lib/debug/observatory_debugger.dart (hunk 3 above)
Comment (johnmccutchan) [inline lib/debug/observatory_debugger.dart]: Done.
